module/canvas/shaders/hit-box-shader.mjs

```diff
@@ -0,0 +1,194 @@
+    {id:"aHalfSize", size:2, normalized: false, type: PIXI.TYPES.FLOAT},
+    {id:"aCorner", size:2, normalized: false, type: PIXI.TYPES.FLOAT}
+  ];
+
+  /** @override */
+  static batchVertexSize = 7;
+
+  /** @override */
+  static batchDefaultUniforms() {
+    return {
+      time: 0,
+      gridSize: 100,
+      thickness: 25,
+      dashCount: 12,
+      dashNormLength: 1,
+      dashBlur: 0.15,
+      dashCyclePerSecond: 1,
+      zoomScale: 1.0
+    };
+  }
+
+  /* -------------------------------------------- */
+
+  /** @override */
+  static _preRenderBatch(batchRenderer) {
+    const u = batchRenderer._shader?.uniforms;
+    if ( !u ) return;
+    u.time = canvas.app.ticker.lastTime / 1000;
+    u.gridSize = canvas?.grid?.size ?? u.gridSize;
+    u.zoomScale = canvas?.stage?.scale.x || 1.0;
+  }
+
+  /* -------------------------------------------- */
+
+  /** @override */
+  static _packInterleavedGeometry(element, attributeBuffer, indexBuffer, aIndex, iIndex) {
+    const {float32View, uint32View} = attributeBuffer;
+
+    // Indices data
+    const baseVertex = aIndex / this.vertexSize;
+    const idx = element.indices;
+    for ( let n = 0; n < idx.length; n++ ) indexBuffer[iIndex++] = baseVertex + idx[n];
+
+    // Token data
+    const token = element.object?.object || element.object;
+    const hb = token.getHitboxCache();
+    const abgr = hb.abgr;
+    const hitboxCenterX = hb.hitboxCenterX;
+    const hitboxCenterY = hb.hitboxCenterY;
+    const hitboxHalfWidth = hb.hitboxHalfWidth;
+    const hitboxHalfHeight = hb.hitboxHalfHeight;
+
+    // The four corners to deduct the viewport from center and grid size
+    const corners = [-1, -1, 1, -1, 1, 1, -1, 1];
+
+    // Packing data
+    const vs = this.vertexSize;
+    for ( let i = 0, j = 0; i < corners.length; i += 2, j += vs ) {
+      let k = aIndex + j;
+      uint32View[k++] = abgr;
+      float32View[k++] = hitboxCenterX;
+      float32View[k++] = hitboxCenterY;
+      float32View[k++] = hitboxHalfWidth;
+      float32View[k++] = hitboxHalfHeight;
+      float32View[k++] = corners[i];
+      float32View[k++] = corners[i + 1];
+    }
+  }
+
+  /* -------------------------------------------- */
+
+  /** @override */
+  static get batchVertexShader() {
+    return `#version 300 es
+    precision ${PIXI.settings.PRECISION_VERTEX} float;
+    
+    in vec2 aCenter;
+    in vec2 aHalfSize;
+    in vec4 aColor;
+    in vec2 aCorner;
+    
+    uniform mat3 projectionMatrix;
+    uniform vec4 tint;
+    
+    out vec2 vLocal;
+    flat out vec2 vHalfSize;
+    flat out vec4 vColor;
+
+    void main() {
+      vec2 pos = aCenter + aCorner * aHalfSize;
+      gl_Position = vec4((projectionMatrix * vec3(pos, 1.0)).xy, 0.0, 1.0);
+     
+      vLocal = aCorner * aHalfSize;
+      vHalfSize = aHalfSize;
+      vColor = aColor * tint;
+    }
+    `;
+  }
+
+  /* -------------------------------------------- */
+
+  /** @override */
+  static get batchFragmentShader() {
```

Comment:
The initial animation is a good prototype, I'm sure we can make it nicer. I think the simplest thing to do as a starting point would be to add a little bit of color variation to it with some gradient of color rather than a solid color. Imagine the shape of the animation but with a sort of radial gradient that gives it a bit of subtle variation.

Reply:
For the moment, I have not added an extra color. This is something we should discuss.